src/main/webapp/app/overview/exercise-details/request-feedback-button/request-feedback-button.component.ts

```diff
@@ -0,0 +1,117 @@
+
+    isGeneratingFeedback = input<boolean>();
+    smallButtons = input<boolean>(false);
+    exercise = input.required<Exercise>();
+    generatingFeedback = output<void>();
+
+    private feedbackSent = false;
+    private profileService = inject(ProfileService);
+    private alertService = inject(AlertService);
+    private courseExerciseService = inject(CourseExerciseService);
+    private translateService = inject(TranslateService);
+    private exerciseService = inject(ExerciseService);
+    private participationService = inject(ParticipationService);
+
+    protected readonly ExerciseType = ExerciseType;
+
+    ngOnInit() {
+        this.profileService.getProfileInfo().subscribe((profileInfo) => {
+            this.athenaEnabled = profileInfo.activeProfiles?.includes(PROFILE_ATHENA);
+        });
+        this.isExamExercise = isExamExercise(this.exercise());
+        if (this.isExamExercise || !this.exercise().id) {
+            return;
+        }
+        this.updateParticipation();
+    }
+
+    private updateParticipation() {
+        if (this.exercise().id) {
+            this.exerciseService.getExerciseDetails(this.exercise().id!).subscribe({
+                next: (exerciseResponse: HttpResponse<ExerciseDetailsType>) => {
+                    this.participation = this.participationService.getSpecificStudentParticipation(exerciseResponse.body!.exercise.studentParticipations ?? [], false);
+                },
+                error: (error: HttpErrorResponse) => {
+                    this.alertService.error(`artemisApp.${error.error.entityName}.errors.${error.error.errorKey}`);
+                },
+            });
+        }
+    }
+
+    requestFeedback() {
+        if (!this.assureConditionsSatisfied()) {
+            return;
+        }
+
+        this.courseExerciseService.requestFeedback(this.exercise().id!).subscribe({
+            next: (participation: StudentParticipation) => {
+                if (participation) {
+                    this.generatingFeedback.emit();
+                    this.feedbackSent = true;
+                    this.alertService.success('artemisApp.exercise.feedbackRequestSent');
+                }
+            },
+            error: (error: HttpErrorResponse) => {
+                this.alertService.error(`artemisApp.exercise.${error.error.errorKey}`);
+            },
+        });
+    }
+
+    /**
+     * Checks if the conditions for requesting automatic non-graded feedback are satisfied.
+     * The student can request automatic non-graded feedback under the following conditions:
+     * 1. They have a graded submission.
+     * 2. The deadline for the exercise has not been exceeded.
+     * 3. There is no already pending feedback request.
+     * @returns {boolean} `true` if all conditions are satisfied, otherwise `false`.
+     */
+    assureConditionsSatisfied(): boolean {
+        if (this.exercise().type !== ExerciseType.PROGRAMMING && this.hasAthenaResultForLatestSubmission()) {
```

Comment:
I'd suggest inverting this conditional.
Typically it's best to have the least damaging value as the fallback value. In this case, this would be false. This is different from gates where you would want to check off every bad case early. However, we're not processing any functionality here.